Extension/src/LanguageServer/client.ts

```diff
@@ -2348,6 +2348,27 @@ export class DefaultClient implements Client {
         return result;
     }
 
+    // This is used to avoid processing unnecessary LSP cancel requests during a Copilot completion
```

Comment:
@sean-mcmanus the exact reason is that caching is happening on copilot-client, hence even if the token is cancelled, it means that the return value will NOT be used for the prompt construction, but it will be used for caching purposes.

Reply:
Yes, I know it's not used for the currently running completion (it doesn't block Copilot), but it's cached and used on the next one. Does my comment not explain that? i.e. should I re-word my comment?

Reply:
I'd omit the 2-minute detail since it is an implementation detail not located in this codebase. Instead, I'd mention that upon cancellation, the result will not be used to build the prompt (since the copilot extension gave up on waiting for a value from this provider), but the returned value is cached by the copilot extension and used on a subsequent call when available, where the cache will be hit and the call to getProjectContext will be saved.

The 2 minutes is the cache entry eviction time defined by the copilot extension.

Reply:
Ok, I tried to re-word it.